### DISCUSSION TITLE
Complement moving boundary in CoupledVarThresholdElementSubdomainModifier

### DISCUSSION THREAD

**Discussion by Wendy-Ji** [idaholab/moose#25736] [Q&A General]
Hello,

I had a question about the complement_moving_boundary parameter in the CoupledVarThresholdElementSubdomainModifier. 

```
[UserObjects]
  [activate_elements]
    type = CoupledVarThresholdElementSubdomainModifier
    coupled_var = 'act_var'
    criterion_type = 'above'
    threshold = 0.5
    subdomain_id = 1 # active block 
    complement_moving_boundary_name = 'moving_interface'
  []
[]
```

In scenario 1 (see image), I am moving elements from the blue subdomain to the grey subdomain, and the green boundary is the complement_moving_boundary. This updates well as the elements move. However, I would like the elements that are activated to move into a different subdomain to the grey one as it will have different properties. 

In scenario 2 (see image), I have started off with some elements already in the red domain, and elements move from the blue to the red. The complement_moving_boundary does update to account for the new red domain, but the part of it previously between the red and grey is not removed. 

I was wondering how I could set the parameters so the complement_moving_boundary doesn't keep parts that are between the red and grey? I am also block restricting my variables to the red and grey domains.

![moose_question_moving_boundary](https://github.com/idaholab/moose/assets/61012460/3b83a206-8fa8-4c36-ba2a-36dcc006f67a)

Thank you.

Wendy

**Comment by lindsayad** [comment]
@hugary1995 it looks like you're the author of this class?

**Reply by hugary1995** [reply]
Yes. 

@Wendy-Ji I'll look into this next week. As I mentioned last time we talked, feel free to tag me directly in the post. I'm committed to a few improvements and bug fixes for this ESM this year.

**Reply by hugary1995** [reply]
I haven't forgotten this thread yet. I was tied to something else in the past weeks. I hope to get to this in the next few days.
